File: mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;
    do
    {
        // ask for height
        printf("Height: ");
        height = get_int();
    }
    // ensure that the height is only between 0 and 23
    while (height < 0 || height > 23);
    {
        for (int rows = 0; rows < height; rows++)
        {
            // print out varying amounts of spaces for each row
            for (int space = height - rows; space - 1 > 0; space--)
            {
                printf(" ");
            }
            // print out varying amounts of hashes for each row
            for (int hash = height - rows; hash - 2 < height; hash++)
            {
                printf("#");
            }
            printf("\n");
        }
    }
}

File: mario.py
from cs50 import get_int

# Make sure the user gives me an integer height between 0 and 23
while True:
    height = get_int("Integer: ")
    if height >= 0 and height <= 23:
        break

# For each row, print a certain number of spaces and hashes
for rows in range(height):
    for space1 in range(height - (rows + 1)):
        print(" ", end="")
    for hash1 in range(rows + 1):
        print("#", end="")
    for space2 in range(2):
        print(" ", end="")
    for hash2 in range(rows + 1):
        print("#", end="")
    print()
